### DISCUSSION TITLE
算法中$$公式环境加号消失

### DISCUSSION THREAD

**Discussion by lingr7** [nuaatug/nuaathesis#73] [Q&A] [answered]
算法环境中加号消失，在单独demo中正常
![Snipaste_2021-11-19_16-35-37](https://user-images.githubusercontent.com/37898999/142592024-e57bcda8-add7-4eef-92cc-98a4a22c1fc3.png)
在模板中，加号消失
![Snipaste_2021-11-19_16-35-49](https://user-images.githubusercontent.com/37898999/142592056-026d3082-a1a1-49c0-bc9d-7c43a53450f4.png)
引入算法宏包代码
```tex
\usepackage[margin=1.5in]{geometry}    % For reducing margin
\usepackage[english]{babel}
\usepackage[utf8]{inputenc}
\usepackage{algorithm}
%\usepackage[algo2e]{algorithm2e} 
\usepackage{arevmath}     % For math symbols
\usepackage{algpseudocode}
```

https://blog.csdn.net/haifeng_gu/article/details/105767750
这里提到了数学相关包mathptmx

**Comment by lingr7** [comment]
https://www.overleaf.com/project/5c49e17e48289d7fe88b2a29
可在这个环境中复现demo效果。
```
\State $ dx \gets x_{i + offset} - mx $
```

**Comment by yzwduck** [comment]
尝试下列方法，但未能复现问题。

1、基于 nuaathesis 里的 demo_chs：
在 global.tex 里添加下列内容：
```latex
\usepackage{algorithm}
\usepackage{algpseudocode}
```
在正文里添加下列内容：
```latex
\begin{algorithmic}
\State $ dx \gets x_{i + offset} - mx $
\end{algorithmic}
```
编译结果能显示 + 号。

2、单独创建文档：
```latex
\documentclass{book}

\usepackage[margin=1.5in]{geometry}    % For reducing margin
\usepackage[english]{babel}
\usepackage[utf8]{inputenc}
\usepackage{algorithm}
%\usepackage[algo2e]{algorithm2e}
%\usepackage{arevmath}     % For math symbols
\usepackage{algpseudocode}

\begin{document}
\begin{algorithmic}
\State $ dx \gets x_{i + offset} - mx $
\end{algorithmic}
\end{document}
```
结果仍然能显示 + 号。
注：尝试过引入 arevmath 包，但没能解决编译问题。

3、访问 Overleaf，结果提示没有权限。

请提供完整的 [Minimum Working Example](https://minimalbeispiel.de/mini-en.html)。没有 MWE 的话，其他人难以调查你遇到的问题。

**Reply by lingr7** [reply]
https://www.overleaf.com/read/rddymzwmnqgb
这个链接应该可以了。

**Comment by lingr7** [comment, marked as answer]
```tex
%\usepackage[margin=1.5in]{geometry}    % For reducing margin
%\usepackage[english]{babel}
\usepackage[utf8]{inputenc}
\usepackage{algorithm}
%\usepackage[algo2e]{algorithm2e} 
%\usepackage{arevmath}     % For math symbols
\usepackage{algpseudocode}
```
注释掉几行之后，能够编译通过并正常显示加号。
![Snipaste_2021-11-21_11-34-00](https://user-images.githubusercontent.com/37898999/142748435-aebb19f1-64f9-4872-b951-e30d39088b48.png)
就是数学字体相关的冲突。现在搞定了。
